src/components/ReservationsCard.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="reservations-card">
+    <div class="reservations-card__wrapper-first">
+      <h2 class="reservations-card__title">Upcoming</h2>
+      <BookingConfirmationCard
+        class="reservations-card__card"
+        v-for="reservation in reservations"
+        :key="reservation.id"
+        :id="reservation.id"
+        :title="reservation.title"
+        :seat="reservation.seat"
+        :datetime="reservation.datetime"
+        :ticketType="reservation.ticketType"
+        :reservationStatus="reservation.reservationStatus"
+        @remove="removeItem"
+        allowRemove
+      />
+    </div>
+    <div class="reservations-card__wrapper-second">
+      <h2 class="reservations-card__title">Past</h2>
+      <BookingConfirmationCard
+        class="reservations-card__card"
+        v-for="reservation in reservations"
+        :key="reservation.id"
+        v-bind="reservation"
```

Comment:
This is clever, I'll give you that 😄
But have you considered having a single prop for the reservation object itself instead of many props for each of it's field?
We could then possibly also introduce typescript sometime later on to make the objects we pass are properly typed and have all the fields we expect 💡